File: 30 days of code( c++)/day14.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

class Difference {
    private:
    vector<int> elements;
  
  	public:
  	int maximumDifference;

	// Add your code here
Difference(vector<int> a )
{
elements = a;
}
void computeDifference(){
 short int i, max = 0, min = 101;
        for(i = 0; i < elements.size(); ++i)
            max = elements[i] > max ? elements[i] : max;
        for(i = 0; i < elements.size(); ++i)
            min = elements[i] < min ? elements[i] : min;
        maximumDifference = max - min;
}
}; // End of Difference class

int main() {
    int N;
    cin >> N;
    
    vector<int> a;
    
    for (int i = 0; i < N; i++) {
        int e;
        cin >> e;
        
        a.push_back(e);
    }
    
    Difference d(a);
    
    d.computeDifference();
    
    cout << d.maximumDifference;
    
    return 0;
}

File: 30 days of code( python)/day14.py
class Difference:
    def __init__(self, a):
        self.__elements = a
        self.maximumDifference = 0

    # Add your code here
    def computeDifference(self):
        l = len(a)
        for i in range(0, l):
            for j in range(i + 1, l):
                difference = abs(a[i] - a[j])
                self.maximumDifference = max(difference, self.maximumDifference)
# End of Difference class

a = [int(e) for e in "1 2 5".split(' ')]

d = Difference(a)
d.computeDifference()

print(d.maximumDifference)